### PR TITLE
Download router config

### LINKED ISSUE #292: ELB OTP deployment failed to assign ec2 server to target group

### LINKED ISSUE #293: Router config not downloaded by OTP ec2 instances that simply download Graph.obj

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

- fix #293 (handle router-config upload/download for instances downloading Graph.obj)
- fix #292 (check that instances properly added to target group)

### CODE REVIEW

File: src/main/java/com/conveyal/datatools/manager/models/Deployment.java [hunk 1]
```diff
@@ -385,6 +374,35 @@ public void dump (File output, boolean includeManifest, boolean includeOsm, bool
...
+            return customRouterConfig != null
+                ? customRouterConfig.getBytes(StandardCharsets.UTF_8)
+                : project.routerConfig != null
+                ? mapper.writer().writeValueAsBytes(project.routerConfig)
+                : null;
+        } catch (JsonProcessingException e) {
+            LOG.error("Router config contains malformed JSON", e);
+            return null;
+        }
+    }
```
Comment (evansiroky) [inline src/main/java/com/conveyal/datatools/manager/models/Deployment.java:404]: Lots of common code here. Seems like another method could be created with just the ObejctMapper writing stuff to bytes part.

**Review by evansiroky — CHANGES_REQUESTED** [review]
Just one thing.

File: src/main/java/com/conveyal/datatools/manager/models/Deployment.java [hunk 2]
```diff
@@ -385,6 +374,40 @@ public void dump (File output, boolean includeManifest, boolean includeOsm, bool
         out.close();
     }
 
+    /** Generate build config for deployment as byte array (for writing to file output stream). */
+    public byte[] generateBuildConfig() {
+        Project project = this.parentProject();
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setSerializationInclusion(Include.NON_NULL);
```
Comment (evansiroky) [inline src/main/java/com/conveyal/datatools/manager/models/Deployment.java:381]: ObjectMapper no longer needed here. Please remove.

**Review by evansiroky — APPROVED** [review]
Good to go after removing some dead code.

**Review by binh-dam-ibigroup — APPROVED** [review]
One small suggestion, good to do.

File: src/main/java/com/conveyal/datatools/manager/jobs/MonitorServerStatusJob.java [hunk 3]
```diff
@@ -147,7 +142,7 @@ public void jobLogic() {
             return;
         }
         graphBuildSeconds = (System.currentTimeMillis() - graphBuildStartTime) / 1000;
-        message = String.format("Graph build/download completed in %d seconds!", graphBuildSeconds);
+        String message = String.format("Graph build/download completed in %d seconds!", graphBuildSeconds);
         LOG.info(message);
```
[the remark this answers is not in the document]
Comment (landonreed) [reply src/main/java/com/conveyal/datatools/manager/jobs/MonitorServerStatusJob.java:146]: Nah, you can't tell in this limited preview, but the same message is used below, so it needs to be formatted with String#format.

**Pull request merged**